packages/components/organisms/f-checkout/src/services/mapper.js

```diff
@@ -49,4 +49,37 @@ const mapUpdateCheckoutRequest = ({
     }
 ]);
 
-export default mapUpdateCheckoutRequest;
+/**
+ * Maps checkout names to required GA names.
+ *
+ * Keys -`checkout` names.
+ * Values - required analytics names.
+ */
+const analyticFieldNameMapper = {
+    'address.line1': 'addressLine1',
+    line1: 'addressLine1',
+    'address.line2': 'addressLine2',
+    line2: 'addressLine2',
+    'address.city': 'addressCity',
+    city: 'addressCity',
+    'address.postcode': 'addressPostcode',
+    postcode: 'addressPostcode',
+    'customer.firstName': 'firstName',
+    'customer.lastName': 'lastName',
+    'customer.mobileNumber': 'phone',
+    mobileNumber: 'phone',
+    'customer.email': 'email'
+};
+
+/**
+ * Updates passed field names to match the expected analytics name requirements.
+ * If passed fields are an array. Sorts the array alphabetically
+ */
+const mapAnalyticsFieldNames = fields => {
+    if (Array.isArray(fields)) {
+        return fields.map(field => (analyticFieldNameMapper[field] ? analyticFieldNameMapper[field] : field)).sort();
+    }
+    return analyticFieldNameMapper[fields] ? analyticFieldNameMapper[fields] : fields;
+};
```

Comment:
As we discussed, I'd probably split this into two separate functions. You have two clearly distinct behaviours within one function currently.

Reply:
Ah yeah, sorry. I'll sort that out